xarray/core/combine.py

```diff
@@ -548,31 +547,142 @@ def _auto_combine(datasets, concat_dims, compat, data_vars, coords,
+    <xarray.Dataset>
+    Dimensions:         (x: 4, y: 4)
+    Coordinates:
+      lon               (x, y) float64 -99.83 -99.32 -99.79 -99.23
+      lat               (x, y) float64 42.25 42.21 42.63 42.59
+    Dimensions without coordinates: x, y
+    Data variables:
+      temperature       (x, y) float64 11.04 23.57 20.77 ...
+      precipitation     (x, y) float64 5.904 2.453 3.404 ...
+
+    See also
+    --------
+    concat
+    merge
+    """
+
+    # The IDs argument tells _combine that the datasets are not yet sorted
+    return _combine(datasets, concat_dims=concat_dims, compat=compat,
+                    data_vars=data_vars, coords=coords,
+                    infer_order_from_coords=False, ids=False)
+
+
 def auto_combine(datasets, concat_dim=_CONCAT_DIM_DEFAULT,
                  compat='no_conflicts', data_vars='all', coords='different'):
-    """Attempt to auto-magically combine the given datasets into one.
-    This method attempts to combine a list of datasets into a single entity by
-    inspecting metadata and using a combination of concat and merge.
-    It does not concatenate along more than one dimension or sort data under
-    any circumstances. It does align coordinates, but different variables on
-    datasets can cause it to fail under some scenarios. In complex cases, you
-    may need to clean up your data and use ``concat``/``merge`` explicitly.
-    ``auto_combine`` works well if you have N years of data and M data
-    variables, and each combination of a distinct time period and set of data
-    variables is saved its own dataset.
+    """
+    Attempt to auto-magically combine the given datasets into one.
+
+    This method attempts to combine a group of datasets along any number of
+    dimensions into a single entity by inspecting coords and metadata and using
+    a combination of concat and merge.
+
+    Will attempt to order the datasets such that their coordinate values are
+    monotonically increasing along all dimensions. If it cannot determine the
+    order in which to concatenate the datasets, it will raise an error.
+
+    Aligns coordinates, but different variables on datasets can cause it
+    to fail under some scenarios. In complex cases, you may need to clean up
+    your data and use concat/merge explicitly (also see `manual_combine`).
+
+    Works well if, for example, you have N years of data and M data variables,
+    and each combination of a distinct time period and set of data variables is
+    saved as its own dataset. Also useful for if you have a simulation which is
+    parallelized in multiple dimensions, but has global coordinates saved in
+    each file specifying it's position within the domain.
+
     Parameters
     ----------
     datasets : sequence of xarray.Dataset
-        Dataset objects to merge.
-    concat_dim : str or DataArray or Index, optional
+        Dataset objects to combine.
+    concat_dim : str, DataArray, Index or None, optional
```

Comment:
I'm not sure if there are any cases in which it makes sense to insert new dimensions when doing automatic combining. Perhaps we should drop this argument?

Reply:
That would mean that a semi-automatic merge-then-concat (e.g. for combining simulations with different parameter choices into one large dataset) which used to be doable in one `auto_combine` now has to be done in two lines (`auto_combine` then `concat`). 

However I think I agree that restricting the scope of the function would be clearer overall though.